### PR TITLE
chore: aws cdk v2 migration troubleshooting

### PR DESCRIPTION
**Pull request by akshbhu**
_Description of changes:_

Updates troubleshooting for cdkv2 migration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/pages/cli/project/troubleshooting.mdx [hunk 1]
```diff
@@ -235,6 +235,22 @@ Amplify currently doesn't support automatic data migration. If the data in the o
 If you have previously deployed an application using GraphQL transformer V1 and have decided to migrate to GraphQL transformer V2, go through the [GraphQL transformer migration documentation](/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you) to understand the impact of the migration on the schema directives and limitations prior to updating.
 You can confirm the version of the GraphQL transformer using the `amplify status` command. You can also inspect the `transformerversion` feature flag in `${project-root}/amplify/cli.json`, to confirm.
 
+### Migration from Amplify version using AWS CDK V1 to V2
+If you have previously deployed an application based on Amplify version < 11.0.0 which is using overrides or custom resources, you can follow the below guide if you are in process of migration amplify version 10 to 11:
```
Comment (jhockett) [inline src/pages/cli/project/troubleshooting.mdx:239]: I suggest simplifying the wording a bit:

```suggestion
### Version 11 migration from AWS CDK V1 to V2
If you have deployed an application which is using overrides or custom resources prior to Amplify CLI version 11.0.0, follow the steps below to migrate:
```

Additionally, is there any CDK documentation we can point that detail any breaking changes? I see this migration guide is in troubleshooting, but I think it makes more sense to have a standalone entry in the `Migration & Backwards Compatibility` section of the docs.

Comment (akshbhu) [reply src/pages/cli/project/troubleshooting.mdx:239]: There is no migration steps involved for most of the users as amplify with `cdkv2` is almost a non breaking change.

I was confused to have it in migration section. This documentation helps when customers manually upgrade their package.json and `.ts` file to upgrade to `cdkv2` and  this link will be in the warning message if Cx are on `cdkV1`.


I have added `cdkv2` documentation at the end .

**Review by jhockett — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Comment by katiegoines** [comment]
Converting to a draft until this is ready to be merged. @akshbhu, @josefaidt  - Please let me know if the `do-not-merge` label is still accurate. Thanks!

**Review by sdstolworthy — DISMISSED** [review]

File: src/pages/cli/project/troubleshooting.mdx [hunk 2]
```diff
@@ -235,6 +235,22 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+1. For Amplify projects using `overrides`:
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0`
+2. For Amplify projects using `custom`:
+   Update `package.json` for `custom` categories as shown below: 
+   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
+   And update `cdk-stack.ts` for `custom` categories as shown below: 
+   ![Amplify custom warning](/images/ts-guide-cdk.png)
+
+If the problem still persists, follow the below link to file a github issue:
+   https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml
```
Comment (edwardfoyle) [inline src/pages/cli/project/troubleshooting.mdx:249]:
```suggestion
If the problem persists, create an issue [here](https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml)
```

File: src/pages/cli/project/troubleshooting.mdx [hunk 3]
```diff
@@ -235,6 +235,22 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+
+To know more about the CDK Migration procedure, follow the below link:
```
Comment (edwardfoyle) [inline src/pages/cli/project/troubleshooting.mdx:251]:
```suggestion
Learn more about the CDK Migration procedure at:
```

File: src/pages/cli/project/troubleshooting.mdx [hunk 4]
```diff
@@ -235,6 +235,22 @@ Amplify currently doesn't support automatic data migration. If the data in the o
 If you have previously deployed an application using GraphQL transformer V1 and have decided to migrate to GraphQL transformer V2, go through the [GraphQL transformer migration documentation](/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you) to understand the impact of the migration on the schema directives and limitations prior to updating.
 You can confirm the version of the GraphQL transformer using the `amplify status` command. You can also inspect the `transformerversion` feature flag in `${project-root}/amplify/cli.json`, to confirm.
 
+### Version 11 migration from AWS CDK V1 to V2
+If you have deployed an application which is using overrides or custom resources prior to Amplify CLI version 11.0.0, follow the steps below to migrate:
```
Comment (ykethan) [inline src/pages/cli/project/troubleshooting.mdx:239]:
```suggestion
If the Amplify project has been deployed with an Amplify CLI version prior to 11.0.0 that utilizes overrides or custom resources, follow the steps below to migrate: 
```

**Review by sdstolworthy — APPROVED** [review]

File: src/pages/cli/project/troubleshooting.mdx [hunk 5]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+If the Amplify project has been deployed with an Amplify CLI version prior to 11.0.0 that utilizes overrides or custom resources, follow the steps below to migrate:
+1. For Amplify projects using `overrides`:
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0`
+2. For Amplify projects using `custom`:
+   Update `package.json` for `custom` categories as shown below: 
+   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
+   And update `cdk-stack.ts` for `custom` categories as shown below: 
+   ![Amplify custom warning](/images/ts-guide-cdk.png)
+
+If the problem persists, create an issue [here](https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml)
```
Comment (sdstolworthy) [inline src/pages/cli/project/troubleshooting.mdx:248]:
```suggestion
If the problem persists, create an issue [here](https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml).
```

**Review by sobolk — APPROVED** [review]

**Review by edwardfoyle — DISMISSED** [review]

**Review by ykethan — APPROVED** [review]

**Comment by akshbhu** [comment]
@aws-amplify/documentation-team  @aws-amplify/amplify-pms

File: src/pages/cli/project/troubleshooting.mdx [hunk 6]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
 If you have previously deployed an application using GraphQL transformer V1 and have decided to migrate to GraphQL transformer V2, go through the [GraphQL transformer migration documentation](/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you) to understand the impact of the migration on the schema directives and limitations prior to updating.
 You can confirm the version of the GraphQL transformer using the `amplify status` command. You can also inspect the `transformerversion` feature flag in `${project-root}/amplify/cli.json`, to confirm.
 
+### Version 11 migration from AWS CDK V1 to V2
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:238]: should this be a separate page in "Migrations and backwards compatibility"?

File: src/pages/cli/project/troubleshooting.mdx [hunk 7]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+If the Amplify project has been deployed with an Amplify CLI version prior to 11.0.0 that utilizes overrides or custom resources, follow the steps below to migrate:
+1. For Amplify projects using `overrides`:
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0`
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:241]: Where do I upgrade this? What file?

File: src/pages/cli/project/troubleshooting.mdx [hunk 8]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+2. For Amplify projects using `custom`:
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:242]:
```suggestion
2. For Amplify projects using `amplify add custom`:
```

File: src/pages/cli/project/troubleshooting.mdx [hunk 9]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+   Update `package.json` for `custom` categories as shown below:
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:243]: Where's this package located?

File: src/pages/cli/project/troubleshooting.mdx [hunk 10]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
 If you have previously deployed an application using GraphQL transformer V1 and have decided to migrate to GraphQL transformer V2, go through the [GraphQL transformer migration documentation](/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you) to understand the impact of the migration on the schema directives and limitations prior to updating.
 You can confirm the version of the GraphQL transformer using the `amplify status` command. You can also inspect the `transformerversion` feature flag in `${project-root}/amplify/cli.json`, to confirm.
 
+### Version 11 migration from AWS CDK V1 to V2
+If the Amplify project has been deployed with an Amplify CLI version prior to 11.0.0 that utilizes overrides or custom resources, follow the steps below to migrate:
+1. For Amplify projects using `overrides`:
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0`
+2. For Amplify projects using `custom`:
+   Update `package.json` for `custom` categories as shown below: 
+   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:244]: Maybe use ```diff annotation instead of a picture

File: src/pages/cli/project/troubleshooting.mdx [hunk 11]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+   And update `cdk-stack.ts` for `custom` categories as shown below:
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:245]: where's the "cdk-stack.ts" file located?

**Review by renebrandel — CHANGES_REQUESTED** [review]
Commented requested changes inline

File: src/pages/cli/project/troubleshooting.mdx [hunk 12]
```diff
@@ -235,6 +235,21 @@ Amplify currently doesn't support automatic data migration. If the data in the o
 If you have previously deployed an application using GraphQL transformer V1 and have decided to migrate to GraphQL transformer V2, go through the [GraphQL transformer migration documentation](/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you) to understand the impact of the migration on the schema directives and limitations prior to updating.
 You can confirm the version of the GraphQL transformer using the `amplify status` command. You can also inspect the `transformerversion` feature flag in `${project-root}/amplify/cli.json`, to confirm.
 
+### Version 11 migration from AWS CDK V1 to V2
+If the Amplify project has been deployed with an Amplify CLI version prior to 11.0.0 that utilizes overrides or custom resources, follow the steps below to migrate:
```
Comment (renebrandel) [inline src/pages/cli/project/troubleshooting.mdx:239]: Also add some context as to "why" you need to migrate. What's changing? Why does it affect the customer?

**Review by sobolk — DISMISSED** [review]

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 13]
```diff
@@ -0,0 +1,40 @@
+export const meta = {
+  title: `AWS CDK v1 to v2 migration`,
+  description: ``,
+};
+
+The version 1 of AWS Cloud Development Kit [CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:6]:
```suggestion
AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+Amplify CLI v11 now uses version 2 of CDK to unblock Customers from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:7]:
```suggestion
Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK version.
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+To learn more about AWS CDK Migration : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:8]:
```suggestion
To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+ - amplify/backend/custom/<resource-name1>/package.json
+ - amplify/backend/custom/<resource-name2>/package.json
+
+Follow this guide here : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
+```
+
+## How do I migrate?
+
+If an Amplify project using `amplify overrides [category]` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:
+1. For Amplify projects using `amplify overrides [category]`:
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:30]:
```suggestion
1. For projects using `amplify overrides [category]`:
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0` in `amplify/backend/package.json`
+2. For Amplify projects using `amplify add custom`:
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:32]:
```suggestion
2. For projects using `amplify add custom`:
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+   Update `package.json` for `custom` categories in `amplify/backend/custom/<customResourceName>/package.json` as shown below: 
+   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
+   And update `amplify/backend/custom/<customResourceName>/cdk-stack.ts` for `custom` categories as shown below: 
+   ![Amplify custom warning](/images/ts-guide-cdk.png)
+
+### Note:
+ `aws-cdk-lib` version should match CDK version that ```@aws-amplify/cli-extensibility-helper``` used as dependency  in `amplify/backend/custom/<customResourceName>/package.json`.
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:39]:
```suggestion
`aws-cdk-lib` version should match the CDK version that `@aws-amplify/cli-extensibility-helper` uses as a dependency  in `amplify/backend/custom/<customResourceName>/package.json`.
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+  You can check the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in `package.json` [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore)
```
Comment (renebrandel) [inline src/pages/cli/migration/aws-cdk-migration.mdx:40]:
```suggestion
You can verify the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in the `package.json` [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore)
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+If an Amplify project using `amplify overrides [category]` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:
+1. For projects using `amplify overrides [category]`:
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0` in `amplify/backend/package.json`
+2. For projects using `amplify add custom`:
+   Update `package.json` for `custom` categories in `amplify/backend/custom/<customResourceName>/package.json` as shown below: 
+   ![Amplify overrides warning](/images/ts-guide-cdk-package.png)
+   And update `amplify/backend/custom/<customResourceName>/cdk-stack.ts` for `custom` categories as shown below: 
+   ![Amplify custom warning](/images/ts-guide-cdk.png)
+
+### Note:
```
Comment (ykethan) [inline src/pages/cli/migration/aws-cdk-migration.mdx:36]: we should use: `<Callout warning> </Callout>` for this. and at the top of file makes sense as its important

**Review by renebrandel — DISMISSED** [review]

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+  description: ``,
+};
+
+AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
+Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
+To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
+## When to migrate
+During deployment, Amplify CLI searches for AWS CDK v1 dependencies in your project and if found, Amplify CLI prints a warning to migrate to AWS CDK v2.
+```
+We detect you are using CDK v1 with custom stacks and overrides.AWS CDK v1 has entered
```
Comment (jhockett) [inline src/pages/cli/migration/aws-cdk-migration.mdx:12]:
```suggestion
We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 22]
```diff
@@ -0,0 +1,36 @@
...
+ maintenance mode on June 1, 2022 
+
+Impacted files.
+ - amplify/backend/package.json
+ Upgrade '@aws-amplify/cli-extensibility-helper' to latest version ^3.0.0
+ 
+ - amplify/backend/custom/<resource-name1>/package.json
+ - amplify/backend/custom/<resource-name2>/package.json
+
+Follow this guide here : https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```
[the remark this answers is not in the document]
Comment (jhockett) [reply src/pages/cli/migration/aws-cdk-migration.mdx:22]:
```suggestion
Follow this guide here: https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```

File: src/pages/cli/project/troubleshooting.mdx [hunk 23]
```diff
@@ -235,6 +235,13 @@ Amplify currently doesn't support automatic data migration. If the data in the o
 If you have previously deployed an application using GraphQL transformer V1 and have decided to migrate to GraphQL transformer V2, go through the [GraphQL transformer migration documentation](/cli/migration/transformer-migration/#changes-that-amplify-cli-will-auto-migrate-for-you) to understand the impact of the migration on the schema directives and limitations prior to updating.
 You can confirm the version of the GraphQL transformer using the `amplify status` command. You can also inspect the `transformerversion` feature flag in `${project-root}/amplify/cli.json`, to confirm.
 
+### Version 11 migration from AWS CDK V1 to V2
+If the Amplify project has been deployed with an Amplify CLI version prior to 11.0.0 that utilizes overrides or custom resources, go through the [AWS CDK migration documentation](/cli/migration/aws-cdk-migration/).
+If the problem persists, create an issue [here](https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml)
```
Comment (josefaidt) [inline src/pages/cli/project/troubleshooting.mdx:240]:
```suggestion
If the problem persists, [create an issue](https://github.com/aws-amplify/amplify-cli/issues/new?assignees=&labels=pending-triage&template=1.bug_report.yaml)
```

File: src/pages/cli/project/troubleshooting.mdx [hunk 24]
```diff
@@ -235,6 +235,13 @@ Amplify currently doesn't support automatic data migration. If the data in the o
...
+
+Learn more about the CDK Migration at:
+   https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
```
[the remark this answers is not in the document]
Comment (josefaidt) [reply src/pages/cli/project/troubleshooting.mdx:243]: Taking it a step further,

```suggestion
[Learn more about migrating to CDK v2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
```

**Review by renebrandel — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by ykethan — APPROVED** [review]

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+
+<Callout>
+AWS CDK version should match the CDK version that cli-extensibility-helper package uses as a dependency in `amplify/backend/custom/customResourceName/package.json`.
+
+
+You can verify the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in the `package.json`  [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore).
+
+</Callout>
+
+AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:14]:
```suggestion
[AWS Cloud Development Kit (CDK) version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on June 1, 2023.
```

Should the link point to the cdk v1 docs?

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 26]
```diff
@@ -0,0 +1,44 @@
+export const meta = {
+  title: `AWS CDK v1 to v2 migration`,
+  description: ``,
+};
+
+<Callout>
+AWS CDK version should match the CDK version that cli-extensibility-helper package uses as a dependency in `amplify/backend/custom/customResourceName/package.json`.
+
+
+You can verify the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in the `package.json`  [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore).
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:10]:
```suggestion
You can verify the `aws-cdk-lib` version that [`@aws-amplify/cli-extensibility-helper`](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore) uses in the `amplify/backend/package.json` file.
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+</Callout>
+
+AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
+Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
+To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
+## When to migrate
+During deployment, Amplify CLI searches for AWS CDK v1 dependencies in your project and if found, Amplify CLI prints a warning to migrate to AWS CDK v2.
+```
+We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered
+ maintenance mode on June 1, 2022
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:21]:
```suggestion
We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered maintenance mode on June 1, 2022 
```

removing line break

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+```
+We detect you are using CDK v1 with custom stacks and overrides. AWS CDK v1 has entered
+ maintenance mode on June 1, 2022 
+
+Impacted files.
+ - amplify/backend/package.json
+ Upgrade '@aws-amplify/cli-extensibility-helper' to latest version ^3.0.0
+ 
+ - amplify/backend/custom/<resource-name1>/package.json
+ - amplify/backend/custom/<resource-name2>/package.json
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:28]:
```suggestion
Impacted files:
- `amplify/backend/package.json`
	Upgrade `@aws-amplify/cli-extensibility-helper` to latest version (`^3.0.0`)
- `amplify/backend/custom/<resource-name-1>/package.json`
- `amplify/backend/custom/<resource-name-2>/package.json`
```

for readability

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+You can verify the `aws-cdk-lib` version that `@aws-amplify/cli-extensibility-helper` uses in the `package.json`  [here](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore).
+
+</Callout>
+
+AWS Cloud Development Kit [CDK version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on 6/1/2023.
+Amplify CLI v11 and above uses CDK v2 to help you move away from deprecated CDK as well as unblock new features that won’t be added to version 1 of CDK.
+To learn more about AWS CDK Migration, visit [Migrating to V2](https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html)
+## When to migrate
+During deployment, Amplify CLI searches for AWS CDK v1 dependencies in your project and if found, Amplify CLI prints a warning to migrate to AWS CDK v2.
+```
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:19]:
```suggestion

```console
```

add language to fenced code block, separate from prose with a newline

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 30]
```diff
@@ -0,0 +1,44 @@
...
+ 
+ - amplify/backend/custom/<resource-name1>/package.json
+ - amplify/backend/custom/<resource-name2>/package.json
+
+Follow this guide here: https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
+```
+
+## How do I migrate?
+
+If an Amplify project using `amplify overrides [category]` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:35]:
```suggestion
If an Amplify project using `amplify override <category>` or `amplify add custom` was deployed with an Amplify CLI version prior to 11.0.0, then you need to follow the steps below to migrate:
```

migrate `[category]` to `<category>` to align with styleguide. Additionally in CLI help outputs the square brackets denote an optional argument/option, which can be confusing

ref https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md#16-bracket-replacement-text-in-codeblocks

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+1. For projects using `amplify overrides [category]`:
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:36]:
```suggestion
1. For projects using `amplify overrides <category>`:
```

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 32]
```diff
@@ -0,0 +1,44 @@
...
+   Upgrade `@aws-amplify/cli-extensibility-helper` to version `^3.0.0` in `amplify/backend/package.json`
+2. For projects using `amplify add custom`:
+   Update `package.json` for `custom` categories in `amplify/backend/custom/<customResourceName>/package.json` as shown below:
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:39]:
```suggestion
   Update `package.json` for `custom` categories in `amplify/backend/custom/<custom-resource-name>/package.json` as shown below:
```

ref https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md#16-bracket-replacement-text-in-codeblocks

File: src/pages/cli/migration/aws-cdk-migration.mdx (hunk 25 above)
Comment (josefaidt) [reply src/pages/cli/migration/aws-cdk-migration.mdx]: cc @akshbhu I've committed this change but not sure if the link should be updated

File: src/pages/cli/migration/aws-cdk-migration.mdx [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+
+<Callout>
+AWS CDK version should match the CDK version that cli-extensibility-helper package uses as a dependency in `amplify/backend/custom/customResourceName/package.json`.
+
+
+You can verify the `aws-cdk-lib` version that [`@aws-amplify/cli-extensibility-helper`](https://www.npmjs.com/package/@aws-amplify/cli-extensibility-helper?activeTab=explore) uses in the `amplify/backend/package.json` file.
+
+</Callout>
+
+[AWS Cloud Development Kit (CDK) version 1](https://docs.aws.amazon.com/cdk/v2/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on June 1, 2023.
```
Comment (josefaidt) [inline src/pages/cli/migration/aws-cdk-migration.mdx:14]:
```suggestion
[AWS Cloud Development Kit (CDK) version 1](https://docs.aws.amazon.com/cdk/v1/guide/home.html) that Amplify CLI uses under the hood has entered maintenance on June 1, 2022 with end of support date scheduled on June 1, 2023.
```

**Review by josefaidt — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**